hydra-node/test/Hydra/NodeSpec.hs

```diff
@@ -151,3 +172,18 @@ recordNetwork node = do
   patchedNode ref = node{hn = Network{broadcast = recordMsg ref}}
 
   queryMsgs = readIORef
+
+recordServerOutputs :: HydraNode SimpleTx IO -> IO (HydraNode SimpleTx IO, IO [ServerOutput SimpleTx])
+recordServerOutputs node = do
+  ref <- newIORef []
+  pure (patchedNode ref, queryMsgs ref)
+ where
+  recordMsg ref x = atomicModifyIORef' ref $ \old -> (old <> [x], ())
+
+  patchedNode ref = node{server = Server{sendOutput = recordMsg ref}}
+
+  queryMsgs = readIORef
```

Comment:
I guess there;s a good opportunity for refactoring this to make it more DRY with the `recordNetwork` just above.